modeshape-jcr/src/main/java/org/modeshape/jcr/mimetype/MimeTypeDetectors.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * ModeShape (http://www.modeshape.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modeshape.jcr.mimetype;
+
+import org.modeshape.common.logging.Logger;
+import org.modeshape.jcr.Environment;
+import org.modeshape.jcr.RepositoryConfiguration;
+import org.modeshape.jcr.mimetype.tika.TikaContentDetector;
+import org.modeshape.jcr.mimetype.tika.TikaNameOnlyDetector;
+
+/**
+ * Class which decides which {@link MimeTypeDetector} implementation to use at runtime
+ * 
+ * @author Horia Chiorean (hchiorea@redhat.com)
+ */
+public final class MimeTypeDetectors {
+    
+    private static final Logger LOGGER = Logger.getLogger(MimeTypeDetector.class);
+    private static boolean TIKA_AVAILABLE; 
+    
+    static {
+        TIKA_AVAILABLE = true;
+        ClassLoader classLoader = MimeTypeDetector.class.getClassLoader();
+        try {
+            Class.forName("org.apache.tika.detect.DefaultDetector", true, classLoader);
+            Class.forName("org.apache.tika.metadata.Metadata", true, classLoader);
```

Comment:
Just to be clear, we require both classes to be present? (Sorry, it's been a long time since I've looked at the Tika stuff.)